device/core/src/client.ts

```diff
@@ -78,19 +76,13 @@ export class Client extends EventEmitter {
    * @param {string}  connStr           A connection string (optional: when not provided, updateSharedAccessSignature must be called to set the SharedAccessSignature token directly).
    * @param {Object}  blobUploadClient  An object that is capable of uploading a stream to a blob.
    */
-  constructor(transport: Client.Transport, connStr?: string, blobUploadClient?: BlobUploadClient) {
+  constructor(transport: Client.Transport, authentication?: string | AuthenticationProvider, blobUploadClient?: BlobUploadClient) {
```

Comment:
>authentication [](start = 43, length = 14)

this code doesn't do anything with the authentication object.  I assume that's coming with the transport changes, right?

Reply:
ah, this is so we don't have a breaking change in the constructor signature.. This is a breaking change in the constructor behavior, however.  Do we want to change this so the connection string can be used to create an authentication object and then pass it into the transport?  That would retain the previous behavior.

---
In reply to: [157028308](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157028308) [](ancestors = 157028308)

Reply:
adding an exception to signal this thing isn't meant to be used by users.

---
In reply to: [157068579](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157068579) [](ancestors = 157068579,157028308)